ui/dashboards/src/components/PanelDrawer/PanelOptionsEditor.tsx

```diff
@@ -23,10 +23,14 @@ export interface PanelOptionsEditorProps {
 export function PanelOptionsEditor(props: PanelOptionsEditorProps) {
   const { kind, value, onChange } = props;
   const { OptionsEditorComponent, createInitialOptions } = usePanelPlugin(kind);
+  const [currentKind, setCurrentKind] = useState(kind);
 
   // When the kind changes, re-init options
   useEffect(() => {
-    onChange(createInitialOptions());
+    if (kind !== currentKind) {
+      onChange(createInitialOptions());
+      setCurrentKind(kind);
+    }
 
     // TODO: See if we can switch up plugin loading so this happens as part of selecting the plugin kind so we don't
     // need this effect at all
```

Comment:
I also thought about moving the `createInitialOptions` into the `onPanelKindChange` in the parent (PanelDrawer), as a way to not have this useEffect. Because at a high-level, we want: `kind` changes ==> change the options. 

But figured I'd leave it here since this comment hints that things might change and plus, maybe we want this part of the logic to be part of this generic editor... like, no matter where PanelOptionsEditor is used, it will ensure that the options are reset correctly (instead of relying on the caller)

Reply:
I think the changes I've made in #559 should take care of this issue (and your other comment about it above). Do you mind if we merge that first, then you can see what impact that has to these changes?

Reply:
Also, feel free to leave comments on that PR if you have a chance to review!

Reply:
That sounds great (re: merge yours in first, see where things land / what impact it has here, and maybe I don't need to add this code)